File: shoring/blackjackEngine/src/test/java/com/panopset/tests/blackjackEngine/AutoTestVerifier.java
package com.panopset.tests.blackjackEngine;

import org.junit.jupiter.api.Assertions;

class AutoTestVerifier {

  int priorHandCount = 0;
  
  void verifyInitialState(int handCount) {
    Assertions.assertEquals(0, handCount);
    Assertions.assertFalse(isHandCountOver1000(handCount));
  }

  void verifyPostAutoRun(int handCount) {
    Assertions.assertTrue(isHandCountOver1000(handCount));
    Assertions.assertEquals(priorHandCount, handCount);
  }

  private boolean isHandCountOver1000(int handCount) {
    return handCount > 1000;
  }
}

File: code/shoring/blackjackEngine/src/test/kotlin/com/panopset/tests/blackjackEngine/AutoTestVerifier.kt
package com.panopset.tests.blackjackEngine

import org.junit.jupiter.api.Assertions

    fun verifyInitialState(handCount: Int) {
        Assertions.assertEquals(0, handCount)
        Assertions.assertFalse(isHandCountOver1000(handCount))
    }

    fun verifyPostAutoRun(priorHandCount: Int, handCount: Int) {
        Assertions.assertTrue(isHandCountOver1000(handCount))
        Assertions.assertEquals(priorHandCount, handCount)
    }

    private fun isHandCountOver1000(handCount: Int): Boolean {
        return handCount > 1000
    }
